source/code/plugins/out_oms_api.rb

```diff
@@ -58,16 +59,23 @@ def post_data(log_type, time_generated_field_name, records)
       headers[OMS::CaseSensitiveString.new("Log-Type")] = log_type
       headers[OMS::CaseSensitiveString.new("x-ms-date")] = Time.now.utc.httpdate()
 
-      azure_resource_id = OMS::Configuration.azure_resource_id
+      azure_resource_id = OMS::Configuration.azure_resource_id if defined?(OMS::Configuration.azure_resource_id)
       if !azure_resource_id.to_s.empty?
-        headers[OMS::CaseSensitiveString.new("x-ms-AzureResourceId")] = OMS::Configuration.azure_resource_id
+        headers[OMS::CaseSensitiveString.new("x-ms-AzureResourceId")] = azure_resource_id
       end
 
-      omscloud_id = OMS::Configuration.omscloud_id
+      omscloud_id = OMS::Configuration.omscloud_id if defined?(OMS::Configuration.omscloud_id)
       if !omscloud_id.to_s.empty?
-        headers[OMS::CaseSensitiveString.new("x-ms-OMSCloudId")] = OMS::Configuration.omscloud_id
+        headers[OMS::CaseSensitiveString.new("x-ms-OMSCloudId")] = omscloud_id
       end
 
+      uuid = OMS::Configuration.get_vm_uuid if defined?(OMS::Configuration.get_vm_uuid)
+      if !uuid.to_s.empty?
+        headers[OMS::CaseSensitiveString.new("x-ms-UUID")] = uuid
+      end
+
+      headers[OMS::CaseSensitiveString.new("X-Request-ID")] = SecureRandom.uuid
```

Comment:
Please log the request id when the request is failed. You may generate the request id in the handle_records method, and pass it to this method, so that you can log it in the rescue OMS::RetryRequestException => e